manifests/params.pp

```diff
@@ -0,0 +1,16 @@
+# == Class agent_upgrade::params
+#
+# This class is meant to be called from agent_upgrade.
+# It sets variables according to platform.
+#
+class agent_upgrade::params {
+  case $::osfamily {
+    'Debian', 'RedHat', 'Amazon': {
```

Comment:
Debian will be coming in a future PR. There's some bits of prep work in this code, but it's not complete or ready yet.